File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/section_3_3_2_ocp/after/PassToStriker.java
package com.grokkingsoftwarearchitecture.chapter03.section_3_3_2_ocp.after;

/**
 * OCP SOLUTION: Open for Extension.
 * * ARCHITECTURE NOTE: This class proves that our system is "Open for extension." 
 * We added this brand new feature (a new play) simply by creating a new file 
 * and implementing the Play interface. We extended the system's capabilities 
 * without touching any existing code!
 */
public class PassToStriker implements Play {
    public void execute() {
        System.out.println("  [Action] Passing the ball to the striker!");
    }
}

File: Chapter 03/C#/Section_3_3_2_OCP/After/PassToStriker.cs
using System;

namespace Chapter03.OCP.After;

/// <summary>
/// OCP SOLUTION: Open for Extension.
/// 
/// ARCHITECTURE NOTE: This class proves that our system is "Open for extension." 
/// We added this brand new feature (a new play) simply by creating a new file 
/// and implementing the IPlay interface. We extended the system's capabilities 
/// without touching any existing code!
/// </summary>
public class PassToStriker : IPlay
{
    public void Execute()
    {
        Console.WriteLine("  [Action] Passing the ball to the striker!");
    }
}